functions/tokens/[[index]].ts

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable import/no-unused-modules */
+import { MetaTagInjector } from '../components/metaTagInjector'
+import getToken from '../utils/getToken'
+
+export const onRequest: PagesFunction = async ({ params, request, next }) => {
+  const { index } = params
+  const networkName = String(index[0]).toUpperCase()
+  let tokenAddress = String(index[1])
+  tokenAddress =
+    tokenAddress !== 'undefined' && tokenAddress === 'NATIVE'
+      ? '0x0000000000000000000000000000000000000000'
+      : tokenAddress
```

Comment:
Instead of using a var (vs const), pull this logic into a `function toTokenAddress` at the top of the file. This has the benefits of (1) being const and (2) being more self-documenting in code.